### ISSUE TITLE
PaddleNLP/examples/model_compression/ofa/run_glue_ofa_depth.py 这个文件在readme没有说明，是干嘛的

### ISSUE THREAD

**Issue by AI-Mart** [PaddlePaddle/PaddleNLP#1240]
PaddleNLP/examples/model_compression/ofa/run_glue_ofa_depth.py 这个文件在readme没有说明，是干嘛的

**Comment by LiuChiachi** [comment]
这个文件比起`run_glue_ofa.py`对增加了对深度进行压缩的部分，实验结果有待补充所以还在README中没有说明。

**Comment by AI-Mart** [comment]
for width_mult in args.width_mult_list:
    # Step8: Broadcast supernet config from width_mult,
    # and use this config in supernet training.
    net_config = utils.dynabert_config(ofa_model, width_mult)
    ofa_model.set_net_config(net_config)
    logits, teacher_logits = ofa_model(
        input_ids, segment_ids, attention_mask=[None, None])
    rep_loss = ofa_model.calc_distill_loss()
    if args.task_name == 'sts-b':
        logit_loss = paddle.zeros(shape=[1], dtype='float32')
    else:
        logit_loss = soft_cross_entropy(logits,
                                        teacher_logits.detach())
    loss = rep_loss + args.lambda_logit * logit_loss
    loss.backward()
optimizer.step()
lr_scheduler.step()
optimizer.clear_grad()

这上面是每种子网络都计算损失，然后这个损失把梯度计算出来后没有更新梯度，到for循环结束，梯度更新，这个更新的梯度是以哪个子网络的损失来更新呢？


for idx, width_mult in enumerate(args.width_mult_list):
    net_config = utils.dynabert_config(ofa_model, width_mult)
    ofa_model.set_net_config(net_config)
    tic_eval = time.time()
    if args.task_name == "mnli":
        acc = evaluate(ofa_model, criterion, metric,
                       dev_data_loader_matched, width_mult)
        evaluate(ofa_model, criterion, metric,
                 dev_data_loader_mismatched, width_mult)
        print("eval done total : %s s" %
              (time.time() - tic_eval))
    else:
        eval_result, eval_loss = evaluate(ofa_model, criterion, metric, dev_data_loader, width_mult)
        print("eval done total : %s s" %
              (time.time() - tic_eval))

    if paddle.distributed.get_rank() == 0:
        best_eval_result = eval_result
        # output_dir = args.output_dir
        output_dir = os.path.join(args.output_dir, "model_%d" % global_step)
        if not os.path.exists(output_dir):
            os.makedirs(output_dir)
        # need better way to get inner model of DataParallel
        model_to_save = model._layers if isinstance(
            model, paddle.DataParallel) else model
        model_to_save.save_pretrained(output_dir)
        tokenizer.save_pretrained(output_dir)

每个子网络都会去评估损失，然后保存模型，那保存的模型是最优子网络的模型参数吗？这里面怎么判断的呢，感觉好像是保存了最后一个子网络的模型参数？


以上，两个问题咨询下，期待回复，谢谢
